sources/packages/backend/workflow/test/2023-2024/parttime-assessment/parttime-assessment-2023-2024-eligibility-CSGD.e2e-spec.ts

```diff
@@ -0,0 +1,75 @@
+import { PROGRAM_YEAR } from "../constants/program-year.constants";
+import {
+  createFakeConsolidatedPartTimeData,
+  executeAssessment,
+} from "../../test-utils";
+import {
+  DependentEligibility,
+  createFakeStudentDependentEligible,
+} from "../../test-utils/factories";
+
+describe(`E2E Test Workflow parttime-assessment-${PROGRAM_YEAR}-eligibility-CSGD.`, () => {
+  it(
+    "Should determine CSGD as eligible when total assessed need is greater than or equal to 1 " +
+      ", total eligible dependant is at least 1 and total family income is less than the threshold.",
+    async () => {
+      // Arrange
+      const assessmentConsolidatedData =
+        createFakeConsolidatedPartTimeData(PROGRAM_YEAR);
+      assessmentConsolidatedData.studentDataDependants = [
+        createFakeStudentDependentEligible(
+          DependentEligibility.Eligible0To18YearsOld,
+        ),
+      ];
+
+      // Act
+      const calculatedAssessment = await executeAssessment(
+        `parttime-assessment-${PROGRAM_YEAR}`,
+        assessmentConsolidatedData,
+      );
+
+      // Assert
+      expect(calculatedAssessment.variables.awardEligibilityCSGD).toBe(true);
+      expect(
+        calculatedAssessment.variables.finalFederalAwardNetCSGDAmount,
+      ).toBeGreaterThan(0);
+    },
+  );
+
+  it("Should determine CSGD as not eligible when total family income is greater than threshold with dependant.", async () => {
+    // Arrange
+    const assessmentConsolidatedData =
+      createFakeConsolidatedPartTimeData(PROGRAM_YEAR);
+    assessmentConsolidatedData.studentDataDependants = [
+      createFakeStudentDependentEligible(
+        DependentEligibility.Eligible0To18YearsOld,
+      ),
+    ];
+    assessmentConsolidatedData.studentDataCRAReportedIncome = 100000;
+
+    // Act
+    const calculatedAssessment = await executeAssessment(
+      `parttime-assessment-${PROGRAM_YEAR}`,
+      assessmentConsolidatedData,
+    );
+
+    // Assert
+    expect(calculatedAssessment.variables.awardEligibilityCSGD).toBe(false);
+  });
+
+  it("Should determine CSGD as not eligible when there is no total eligible dependant.", async () => {
+    // Arrange
+    const assessmentConsolidatedData =
+      createFakeConsolidatedPartTimeData(PROGRAM_YEAR);
+    // No total eligible dependant.
+    assessmentConsolidatedData.studentDataDependants = undefined;
+    // Act
+    const calculatedAssessment = await executeAssessment(
+      `parttime-assessment-${PROGRAM_YEAR}`,
+      assessmentConsolidatedData,
+    );
+
+    // Assert
+    expect(calculatedAssessment.variables.awardEligibilityCSGD).toBe(false);
```

Comment:
asser `calculatedAssessment.variables.finalFederalAwardNetCSGDAmount` to be undefined or null